src/site/lab8.rst

```diff
@@ -0,0 +1,148 @@
+******
+Lab #8
+******
+
+* Refer to the topic pages for this lab
+    * You will find many of the answers to your questions there
+* Feel free to use your laptop if you have it
+* I strongly encourage you to work with others in the lab
+* Ensure I have recorded your completion. Failure to do so will result in a grade of 0
+* When you get stuck, do me a favour and ask those sitting around you for help
+    * I want people to get used to working together in the labs
+    * Peer teaching and peer learning is super effective
+
+
+Recursion Visualization
+=======================
+
+* Check out the following websites that help you visualize recursion:
+    * https://visualgo.net/en/recursion?slide=1
+    * https://cscircles.cemc.uwaterloo.ca/java_visualize/
+
+* Try running the following in the visualizer
+
+    .. code-block:: python
+        :linenos:
+
+        public class Recursion {
+            public static int sum(int n) {
+                if (n == 1) {
+                    return 1;
+                }
+                return n + sum(n - 1);
+            }
+
+            public static void main(String[] args) {
+                sum(5);
+            }
+        }
+
+
+Sum
+===
+
+1. Write an **iterative** method to sum the numbers between ``1`` and ``n`` inclusively
+    * The method will take a single integer ``n`` as a parameter
+    * The method will return the sum
+
+2. Write a **recursive** method to sum the numbers between ``1`` and ``n`` inclusively
+    * The method will take a single integer ``n`` as a parameter
+    * The method will return the sum
+
+3. Write a **recursive** method to sum the numbers between ``cur`` and ``n`` inclusively
+    * The method will take two parameters ``cur`` and ``n``
+    * The method will return the sum
+
+
+Factorial
+=========
```

Comment:
Maybe a one liner describing this, I imagine not everyone in first year will remember what this is. Or a [lmgt](https://letmegooglethat.com/?q=factorial) link would work.

Reply:
Defo going with lmgt